geode-for-redis/src/test/java/org/apache/geode/redis/internal/data/collections/SizeableByteArrayListTest.java

```diff
@@ -58,6 +58,24 @@ public void getSizeInBytesIsAccurate_ForSizeableByteArrayListElements() {
     assertThat(list.size()).isEqualTo(0);
   }
 
+  @Test
+  public void clearSublist_getSizeInBytesIsAccurate() {
+    // Create a list with an initial size and confirm that it correctly reports its size
+    SizeableByteArrayList list = createList();
+    assertThat(list.getSizeInBytes()).isEqualTo(sizer.sizeof(list));
+
+    // Remove subset of elements and assert that the size is correct after each remove
+    Random rand = new Random();
+    while (list.size() > 3) {
+      int fromIndex = rand.nextInt(list.size() / 2);
+      int toIndex = rand.nextInt(list.size() / 2) + fromIndex;
```

Comment:
This test doesn't need to use randomness; that just complicates things and could potentially lead to a more difficult to diagnose bug if the test does fail. For this test, it would be enough to just create a list, remove some sublist from the middle once, then assert that the size is correct.

Reply:
Deterministic now.